Src/FluentAssertions/Formatting/FormattedObjectGraph.PossibleMultilineFragment.cs

```diff
@@ -0,0 +1,91 @@
+﻿using System;
+
+namespace FluentAssertions.Formatting;
+
+/// <summary>
+/// Implementation of the contained PossibleMultilineFragment.
+/// </summary>
+public partial class FormattedObjectGraph
+{
+    /// <summary>
+    /// Write fragments that may be on a single line or span multiple lines,
+    /// and this is not known until later parts of the fragment are written.
+    /// </summary>
+    internal record PossibleMultilineFragment
+    {
+        private readonly FormattedObjectGraph formattedGraph;
+        private readonly int startingLineBuilderIndex;
+        private readonly int startingLineCount;
+
+        public PossibleMultilineFragment(FormattedObjectGraph formattedGraph)
+        {
+            this.formattedGraph = formattedGraph;
+            startingLineBuilderIndex = formattedGraph.lineBuilder.Length;
+            startingLineCount = formattedGraph.lines.Count;
+        }
+
+        /// <summary>
+        /// Write the fragment at the position the graph was in when this instance was created.
+        ///
+        /// <para>
+        /// If more lines have been added since this instance was created then write the
+        /// fragment on a new line, otherwise write it on the same line.
+        /// </para>
+        /// </summary>
+        internal void AddFragmentAtStart(string fragment)
```

Comment:
🤔 `AddStartingFragment`?

Reply:
Heh, yes I went back and forth between `AddFragmentAtStart` and `AddStartingFragment` a number of times.